services/ingestion/etl/budg/xls/src/lib/transform.js

```diff
@@ -0,0 +1,79 @@
+/*
+ * Transform message
+ */
+
+/*
+ * Map fields
+ */
+export default record => {
+  // Preprocess results
+  const resultObject = {
+    available: record['Results Available'],
+    result: record['Results Platform Project Card'],
+  };
+
+  // Preprocess locations
+  const locationArray = record['Participating countries']
+    .split(',')
+    .map((country, index) => ({
+      name: country,
+      geolocation: {
+        lat: null,
+        long: null,
+      },
+    }));
+
+  // Preprocess type
+  const typeArray =
+    (record['Activity type'] != null && record['Activity type'].split(','))
+    || null;
+
+  // Preprocess coordinators
+  const coordArray = [{
+    name: record["Coordinator's name"],
+    type: record['Coordinator organisation type'],
+    address: record["Coordinator's address"],
+    region: record["Coordinator's region"],
+    country: record["Coordinator's country"],
+    website: record["Coordinator's website"],
+  }];
+
+  // Preprocess partners
+  const partnerArray = [];
+  for (let i = 1; i < 39; i++) {
```

Comment:
Where does this `39` come from?

Reply:
It simply goes from the fact that there are 38 columns for partners in their xls file... (why 38 and not something else, I don't know)